File: dotnet/test/Azure.Iot.Operations.Protocol.MetlTests/TestCaseSerializer.cs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

namespace Azure.Iot.Operations.Protocol.MetlTests
{
    public class TestCaseSerializer
    {
        public static string? DefaultOutContentType;
        public static List<string> DefaultAcceptContentTypes = new();
        public static bool DefaultIndicateCharacterData;
        public static bool DefaultAllowCharacterData;
        public static bool DefaultFailDeserialization;

        public string? OutContentType { get; set; } = DefaultOutContentType;

        public List<string> AcceptContentTypes { get; set; } = DefaultAcceptContentTypes;

        public bool IndicateCharacterData { get; set; } = DefaultIndicateCharacterData;

        public bool AllowCharacterData { get; set; } = DefaultAllowCharacterData;

        public bool FailDeserialization { get; set; } = DefaultFailDeserialization;
    }
}

File: go/test/protocol/test_case_serializer.go
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
package protocol

import (
	"gopkg.in/yaml.v3"
)

type testCaseSerializer struct {
	OutContentType        *string  `yaml:"out-content-type"`
	AcceptContentTypes    []string `yaml:"accept-content-types"`
	IndicateCharacterData bool     `yaml:"indicate-character-data"`
	AllowCharacterData    bool     `yaml:"allow-character-data"`
	FailDeserialization   bool     `yaml:"fail-deserialization"`
}

type TestCaseSerializer struct {
	testCaseSerializer
}

func (serializer *TestCaseSerializer) UnmarshalYAML(node *yaml.Node) error {
	*serializer = TestCaseDefaultSerializer.GetSerializer()

	return node.Decode(&serializer.testCaseSerializer)
}
